parser-typechecker/src/Unison/Runtime/Builtin.hs

```diff
@@ -1750,6 +1750,23 @@ declareForeigns = do
   declareForeign "Bytes.fromBase64" boxToEBoxBox . mkForeign $ pure . Bytes.fromBase64
   declareForeign "Bytes.fromBase64UrlUnpadded" boxDirect . mkForeign $ pure . Bytes.fromBase64UrlUnpadded
 
+  declareForeign "Bytes.decodeNat64be" boxToMaybeBox . mkForeign $ pure . (\n -> let bs = Bytes.decodeNat64be n in
+                                                                           traceShow bs bs)
+  declareForeign "Bytes.decodeNat64le" boxToMaybeBox . mkForeign $ pure . Bytes.decodeNat64le
```

Comment:
`boxToMaybeBox` is not going to work here.

Unison tuples are like cons lists `(a,b) = Cons a (Cons b ())`. And the way that the foreign operation is going to pass things back is like this:

  - Tag word indicating `Nothing` vs `Just`
  - In the latter case, word for the `Nat` on the unboxed stack and one closure for the `Bytes` on the boxed stack

So the wrapper code needs to read that stack convention and build the Unison values up.